code/game/machinery/poolcontroller.dm

```diff
@@ -107,25 +106,29 @@
 /obj/machinery/poolcontroller/proc/wires()
 	return wires.GetInteractWindow()
 
-//herpderp	/obj/machinery/poolcontroller/proc/poolreagent()
-//herpderp		for(var/turf/simulated/pool/water/W in linkedturfs)
-//herpderp			for(var/mob/living/carbon/human/swimee in W)
-//herpderp				if(beaker && cur_reagent)
-//herpderp					swimee.reagents.add_reagent(cur_reagent, 1)
-//herpderp					reagenttimer = 5
-//				world << "got [cur_reagent]"
+/obj/machinery/poolcontroller/proc/poolreagent()
+	for(var/turf/simulated/pool/water/W in linkedturfs)
+		for(var/mob/living/carbon/human/swimee in W)
```

Comment:
Chems actually affect all mob/living, I would change this for more variety/killing Ian